command/agent/cache/crypto/k8s_test.go

```diff
@@ -0,0 +1,142 @@
+package crypto
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+)
+
+func TestCrypto_KubernetesNewKey(t *testing.T) {
+	k8sKey, err := NewK8s([]byte{})
+	if err != nil {
+		t.Fatalf(fmt.Sprintf("unexpected error: %s", err))
+	}
+
+	key := k8sKey.GetKey()
+	if key == nil {
+		t.Fatalf(fmt.Sprintf("key is nil, it shouldn't be: %s", key))
+	}
+
+	persistentKey := k8sKey.GetPersistentKey()
+	if persistentKey == nil {
+		t.Fatalf(fmt.Sprintf("key is nil, it shouldn't be: %s", persistentKey))
+	}
```

Comment:
Could check if key == persistentKey here as well, since they should be the same?

Reply:
Done!